### ISSUE TITLE
How to integrate `CompositionalMetric` into a `LightningModule`?

### ISSUE THREAD

**Issue by celsofranssa** [Lightning-AI/metrics#1108]
Currently, I am computing two metrics at the `validation_epoch_end`, as shown in the code snippet below.

```python
class LTModel(pl.LightningModule):

    def __init__(self, hparams):
        super(LTModel, self).__init__()
        self.entity_cls_head = # entity classifier 
        self.relation_cls_head = # relation classifier

        # metrics
        self.entity_metrics = self._get_metrics(prefix="entity_")
        self.relation_metrics = self._get_metrics(prefix="relation_")

        # loss
        self.loss = ...

    def _get_metrics(self, prefix):
        pass
        return MetricCollection(
             metrics={
                 "Wei-F1": F1(num_classes=self.hparams.num_entities, average="weighted"),
                 ...
             },
             prefix=prefix)
```

However, I would like to aggregate these two metrics to form a single one using the [CompositionalMetric](https://github.com/Lightning-AI/metrics/blob/376bc7d3716f85bb6749f1513827fde05257c6b0/src/torchmetrics/metric.py#L845-L953)?
Then, is there an example I can use to achieve this goal?

**Comment by celsofranssa** [comment]
Currently, I implemented a custom metric combining entity and relation metrics (as each of them has its own `preds` and `targets`)

```python
from torchmetrics import Metric, F1

class FERMetric(Metric):
    def __init__(self, params):
        super().__init__()
        self.entity_metric = F1(num_classes=params.num_entities, average=params.average)
        self.relation_metric = F1(num_classes=params.num_relations, average=params.average)

    def update(self, entity_pred, entity_true, relation_pred, relation_true):
        self.entity_metric.update(entity_pred, entity_true)
        self.relation_metric.update(relation_pred, relation_true)

    def compute(self):
        return 0.5 * (
                self.entity_metric.compute() + self.relation_metric.compute()
        )
```
However, it seems not right:
```python
Epoch 6:  36%|██               | ... entity_Mac-F1=0.588, relation_Mac-F1=0.477, val_FER=0.260]
```
since `0.5 * (entity_Mac-F1 + relation_Mac-F1)` should be equal to `val_FER` .
What am I missing?
